New features, bug fix

KavehYousefi opened the pull request:
Added:
- Named captured groups, similar to those in Java and Perl. These follow the pattern "(!&lt;GROUP-NAME&gt;...)". Examples and elucidations will follow soon.
- Ranged quantifiers, similar to Java and Perl. These follow the pattern "{[min,max]}", "{=comma-separated-numbers}" and "{%:user-defined-checking-function:}". Examples and elucidations will follow soon.
- Implemented the means to configure and deactivate the above features to obviate compatibility issues with extant software using the "re" library. Examples and elucidations will follow soon.

NOTES: As mentioned above, these two new features might change the behavior of existing programs based on the "re" library, for instance by interpretation of the "{" character as ranged qualifier start marker.

Fixed:
- In a character set using the character pair "(?" only the "(" character is recognized, as the "?" is processed and lost by the ":group" parsing. Example: (match-re "[(?]" "?") accepts the input string "(", but does not match the input string "?".

Commit:
New features, bug fix

Added:
- Named captured groups, similar to those in Java and Perl.
- Ranged quantifiers, similar to Java and Perl.

Fixed:
- In a character set using the character pair "(?" only the "("
character is recognized, as the "?" is processed and lost by the
":group" parsing.

Commit:
Expansion of the documentation

Added a preliminary description of named captured groups.

massung commented:
Will take a look at this soon. _VERY_ nice! And thank you for updating the README documentation as well.

Commit:
Description of “$->”

Added a description of the local “$->” function.

Commit:
Corrected keyword parameters by adding colons

Commit:
Corrected keyword parameters by appending colons

Commit:
Introduced section about configuration

Commit:
Description of ranged quantifiers

Commit:
Added examples for the ranged quantifiers

Commit:
Inserted prompt snippets and examples

- The prompt prefix “CL-USER > ” has been added to the examples.
- Further examples are offered.
- Some grammatical mistakes have been rectified.

Commit:
Further examples

Added further examples.

Commit:
Moved description of “match-extract-data”

The description of the “match-extract-data” arguments has been moved to the “Basic Pattern Matching” section.

Commit:
Extended the “match-extract-data”

Extended the “match-extract-data” example in the “Basic Pattern Matching” section to include named and unnamed captured groups for more lucid demonstration purposes.

Commit:
Removed erroneous percent sign

The superfluous second percent sign in `{%:USER-FUNCTION%:}` has been removed to yield the correct `{%:USER-FUNCTION:}` syntax definition.

Commit:
Remove unused variable

Updated to the correct last state. This involved removing an unused local variable `$<>`.

Commit:
Amended comment for groups
